src/data_manager.py

```diff
@@ -0,0 +1,110 @@
+import hashlib
+import os
+import pickle
+import sys
+from collections import UserDict
+from models import AddressBook
+
+
+class DataManager:
```

Comment:
- Rename to `DataSerializer`
- Use `DataSerializer` class specifically for serializing data: leave `load_data` and `save_data` methods
- Move `generate_hash/login/signup/validate_secret` to another class - `AuthManager`
- Don't use ``